### PR TITLE
ci: update Go 1.21 support

### PR DESCRIPTION
**Pull request by michalbiesek**

### CODE REVIEW

File: .github/workflows/validate.yml [hunk 1]
```diff
@@ -33,7 +33,7 @@ jobs:
       - uses: actions/checkout@v3
       - uses: actions/setup-go@v3
         with:
-          go-version: 1.20.x
+          go-version: 1.21.x
       - uses: golangci/golangci-lint-action@v3
         with:
           version: v1.51
```
[the remark this answers is not in the document]
Comment (michalbiesek) [reply .github/workflows/validate.yml]: Done.

**Comment by thaJeztah** [comment]
Looks like the updated linter doesn't like the unused (but named) arguments in the stubs.

I guess we can either update the linter settings (in golangci-lint.yml) or remove the names

Not sure what's best here; having the names could still be useful to understand the arguments that are passed to the stubs, but perhaps that's not a big issue;

```
  Warning: unused-parameter: parameter 'options' seems to be unused, consider removing or renaming it as _ (revive)
  Warning: unused-parameter: parameter 'options' seems to be unused, consider removing or renaming it as _ (revive)
  Warning: unused-parameter: parameter 'path' seems to be unused, consider removing or renaming it as _ (revive)
  Warning: unused-parameter: parameter 'fileLabel' seems to be unused, consider removing or renaming it as _ (revive)
  Warning: unused-parameter: parameter 'path' seems to be unused, consider removing or renaming it as _ (revive)
  Warning: unused-parameter: parameter 'label' seems to be unused, consider removing or renaming it as _ (revive)
  Warning: unused-parameter: parameter 'label' seems to be unused, consider removing or renaming it as _ (revive)
  Warning: unused-parameter: parameter 'label' seems to be unused, consider removing or renaming it as _ (revive)
  Warning: unused-parameter: parameter 'fpath' seems to be unused, consider removing or renaming it as _ (revive)
  Warning: unused-parameter: parameter 'class' seems to be unused, consider removing or renaming it as _ (revive)
  Warning: unused-parameter: parameter 'fpath' seems to be unused, consider removing or renaming it as _ (revive)
  Warning: unused-parameter: parameter 'fpath' seems to be unused, consider removing or renaming it as _ (revive)
  Warning: unused-parameter: parameter 'fpath' seems to be unused, consider removing or renaming it as _ (revive)
  Warning: unused-parameter: parameter 'fpath' seems to be unused, consider removing or renaming it as _ (revive)
  Warning: unused-parameter: parameter 'label' seems to be unused, consider removing or renaming it as _ (revive)
  Warning: unused-parameter: parameter 'pid' seems to be unused, consider removing or renaming it as _ (revive)
  Warning: unused-parameter: parameter 'val' seems to be unused, consider removing or renaming it as _ (revive)
  Warning: unused-parameter: parameter 'source' seems to be unused, consider removing or renaming it as _ (revive)
  Warning: unused-parameter: parameter 'sourceRange' seems to be unused, consider removing or renaming it as _ (revive)
  Warning: unused-parameter: parameter 'fd' seems to be unused, consider removing or renaming it as _ (revive)
  Warning: unused-parameter: parameter 'label' seems to be unused, consider removing or renaming it as _ (revive)
  Warning: unused-parameter: parameter 'label' seems to be unused, consider removing or renaming it as _ (revive)
  Warning: unused-parameter: parameter 'label' seems to be unused, consider removing or renaming it as _ (revive)
  Warning: unused-parameter: parameter 'mode' seems to be unused, consider removing or renaming it as _ (revive)
  Warning: unused-parameter: parameter 'label' seems to be unused, consider removing or renaming it as _ (revive)
  Warning: unused-parameter: parameter 'val' seems to be unused, consider removing or renaming it as _ (revive)
  Warning: unused-parameter: parameter 'src' seems to be unused, consider removing or renaming it as _ (revive)
  Warning: unused-parameter: parameter 'fpath' seems to be unused, consider removing or renaming it as _ (revive)
  Warning: unused-parameter: parameter 'src' seems to be unused, consider removing or renaming it as _ (revive)
  Warning: unused-parameter: parameter 'user' seems to be unused, consider removing or renaming it as _ (revive)
```

**Comment by michalbiesek** [comment]
@rhatdan 
Done. I have removed the unused parameters.

**Review by thaJeztah — APPROVED** [review]
LGTM if CI is happy 😄 

thanks!

**Comment by thaJeztah** [comment]
@rhatdan could you re-LGTM this one?

**Comment by rhatdan** [comment]
LGTM

**Comment by michalbiesek** [comment]
@thaJeztah @rhatdan just FYI: The new CI jobs for Go 1.21.x are not marked as `required`.
Screenshot from 2023-08-14 19-35-46

**Pull request merged**